lib/datasource/crate/index.ts

```diff
@@ -141,7 +141,7 @@ async function fetchRegistryInfo(
   let url: URL;
   try {
     url = new URL(registryUrl);
-  } catch (err) {
+  } catch (err) /* istanbul ignore next */ {
```

Comment:
this should have a test
```suggestion
  } catch (err) {
```

Reply:
I guess it shouldn't happen and can be removed if we validate and URL in the index though?

Reply:
if it's validated before, we should remove the catch and let bubble any error through upper layers